pkg/pillar/cmd/zedagent/handleProfile.go

```diff
@@ -0,0 +1,259 @@
+	}
+	if localProfile != nil {
+		log.Function("Using saved local profile")
+		getconfigCtx.readSavedLocalProfile = true
+		return validateAndSetLocalProfile(localProfile, getconfigCtx)
+	}
+	return nil
+}
+
+//prepareLocalProfileServerMap process configuration of network instances to find match with defined localServerURL
+//returns the srcIP for the zero or more network instances on which the locaclProfileServer might be hosted
+//based on a subnet inclusion match or hostname in dns records
+//in form actual link (with hostname replaced) -> source IP
+func prepareLocalProfileServerMap(localServerURL string, getconfigCtx *getconfigContext) (map[string]net.IP, error) {
+	res := make(map[string]net.IP)
+
+	netInstanses := getconfigCtx.pubNetworkInstanceConfig.GetAll()
+	u, err := url.Parse(localServerURL)
+	if err != nil {
+		return nil, fmt.Errorf("checkAndPrepareLocalIP: url.Parse: %s", err)
+	}
+	localProfileServerHostname := u.Hostname()
+	localProfileServerIP := net.ParseIP(localProfileServerHostname)
+	for _, entry := range netInstanses {
+		config := entry.(types.NetworkInstanceConfig)
+		if localProfileServerIP != nil {
+			//check if defined ip of localServer is on subnet
+			if config.Subnet.Contains(localProfileServerIP) {
+				res[localServerURL] = config.Gateway
+			}
+		} else {
+			//check if defined host is in DNS records
+			for _, el := range config.DnsNameToIPList {
+				if el.HostName == localProfileServerHostname {
+					for _, ip := range el.IPs {
+						localServerURLReplaced := strings.Replace(localServerURL, localProfileServerHostname,
+							ip.String(), 1)
+						res[localServerURLReplaced] = config.Gateway
+						log.Functionf(
+							"prepareLocalProfileServerMap: will use url with replaced hostname: %s",
+							localServerURLReplaced)
+					}
+				}
+			}
+		}
+	}
+	return res, nil
+}
+
+// getLocalProfileConfig connects to local profile server to fetch current profile
+func getLocalProfileConfig(localServerURL string, iteration int, getconfigCtx *getconfigContext) error {
+
+	log.Tracef("getLocalProfileConfig(%s, %d)", localServerURL, iteration)
+
+	localServerMap, err := prepareLocalProfileServerMap(localServerURL, getconfigCtx)
+	if err != nil {
+		return fmt.Errorf("getLocalProfileConfig: prepareLocalProfileServerMap: %s", err)
+	}
+
+	if len(localServerMap) == 0 {
+		return fmt.Errorf(
+			"getLocalProfileConfig: cannot find any configured local subnets for localServerURL: %s",
+			localServerURL)
+	}
+
+	var errList []string
+	for link, srcIP := range localServerMap {
+		resp, contents, err := zedcloud.SendLocal(zedcloudCtx, link, srcIP, 0, nil)
+		if err != nil {
+			errList = append(errList, fmt.Sprintf("SendLocal: %s", err))
+			continue
+		}
+		if resp.StatusCode == http.StatusOK {
+			if err := validateProtoMessage(link, resp); err != nil {
+				// send something to ledmanager ???
+				errList = append(errList, fmt.Sprintf("validateProtoMessage: resp header error: %s", err))
+				continue
+			}
```

Comment:
Stlystically odd in that you have a mixture of
if error condition {
     continue
}
if success condition {
    ..
    return
}

Makes sense to structure as if error condition throughout so that the success return is at the end of the loop.